confluent_kafka/avro/__init__.py

```diff
@@ -29,23 +29,23 @@ class AvroProducer(Producer):
     def __init__(self, config, default_key_schema=None,
                  default_value_schema=None, schema_registry=None):
 
-        schema_registry_url = config.pop("schema.registry.url", None)
-        schema_registry_ca_location = config.pop("schema.registry.ssl.ca.location", None)
-        schema_registry_certificate_location = config.pop("schema.registry.ssl.certificate.location", None)
-        schema_registry_key_location = config.pop("schema.registry.ssl.key.location", None)
+        sr_conf = {key.replace("schema.registry.", ""): value
+                   for key, value in config.items() if key.startswith("schema.registry.")}
+
+        if config.get("schema.registry.basic.auth.credentials.source") == 'SASL_INHERIT':
```

Comment:
oh never mind, this is a producer specialized for Avro, forgot that's how it's implemented in this client. I was mirroring here some thoughts from another client where that is not the case.

Reply:
Yeah it doesn't get much more coupled than specialized client intended solely for use with the cached schema registry client. In the long term this will just be a convenience wrapper over a client that sets `key.serializer`, `value.serializer` for you. The Avro serializer will leverage its own configure method much like that of the java clients